src/golioth_coap_client.c

```diff
@@ -830,6 +831,7 @@ static golioth_status_t coap_io_loop_once(
     if (!client->session_connected) {
         // Transitioned from not connected to connected
         GLTH_LOGI(TAG, "Golioth CoAP client connected");
+        golioth_sys_client_connected(client);
```

Comment:
The Zephyr log backend (which lives in the `port` directory) needs the `client` pointer in order to call `golioth_log_*` functions. I don't particularly like this pattern, but the others that I considered felt either less clean or more involved, so I picked this as the "least bad" option, and I'm open to feedback/suggestions on another way to get that pointer into the log module. My biggest concern is that I don't think the platform port should have to be aware of the Golioth system at all. The other options I considered were:
- Changing the log module so it doesn't need a client pointer passed in on every call - this is a bigger architectural change, and I like that the client context is a function argument - it makes the code easier to unit test (less global state).
- Having the platform ports register a callback - this doesn't address the fundamental question of whether the platform port should even know about the golioth service and is more complicated.